src/script.js

```diff
@@ -0,0 +1,346 @@
+    this.button = document.createElement('div');
+    this.bill = document.createElement('div');
+    this.language = '';
+    this.shiftStatus = false;
+    this.CapsLockStatus = false;
+  }
+
+  createAreas() {
+    document.body.append(this.virtualDisplay);
+    this.virtualDisplay.classList.add('virtualDisplay');
+    this.virtualDisplay.append(this.printedText);
+    this.virtualDisplay.append(this.currentLanguage);
+    this.printedText.classList.add('printedText');
+    this.currentLanguage.classList.add('currentLanguage');
+    this.currentLanguage.innerText = this.language;
+    document.body.append(this.myKeyboard);
+    this.myKeyboard.className = 'keyboard';
+    document.body.append(this.info);
+    this.info.classList.add('info');
+    this.info.innerHTML = 'Смена языка: ShiftLeft + ControlLeft. На виртуальной клавиатуре сначала нажать ShiftLeft, затем - ConrolLeft. <br /> Создано в ОС Windows <br /> Поведение клавиш SHIFT и CAPSLOCK в виртуальном режиме имитирует реальное поведение';
+    document.body.append(this.bill);
+    this.bill.classList.add('bill');
+  }
+
+  createButtons() {
+    for (let i = 0; i < buttonsLength.length; i += 1) {
+      const button = document.createElement('div');
+      button.classList.add('button');
+      button.setAttribute('id', codesArray[i]);
+      button.style.width = `${buttonsLength[i]}px`;
+      this.myKeyboard.append(button);
+    }
+  }
+
+  addButtonText() {
+    const allButtons = this.myKeyboard.querySelectorAll('.button');
+
+    for (let x = 0; x < this.ALL_KEYS.length; x += 1) {
+      const button = allButtons[x];
+      if (Array.isArray(this.ALL_KEYS[x])) {
+        button.classList.add('doubleInputButton');
+        const noShiftText = document.createElement('div');
+        const shiftText = document.createElement('div');
+        shiftText.classList.add('shiftText');
+        noShiftText.classList.add('noShiftText');
+        button.append(shiftText);
+        button.append(noShiftText);
+        const [a, b] = this.ALL_KEYS[x];
+        shiftText.innerHTML = b;
+        noShiftText.innerHTML = a;
+      } else {
+        button.innerHTML = this.ALL_KEYS[x];
+      }
+    }
+  }
+
+  addListeners() {
+    document.addEventListener('keydown', (event) => this.RealKeyDown(event));
+    document.addEventListener('keyup', (event) => this.RealKeyUp(event));
+    document.addEventListener('mousedown', (event) => this.MouseDown(event));
+    document.addEventListener('mouseup', (event) => this.MouseUp(event));
+    document.addEventListener('mouseout', (event) => this.MouseOut(event));
+  }
+
+  RealKeyDown(event) {
+    event.preventDefault();
+    if (!codesArray.includes(event.code)) { return; }
+
+    if (event.code === 'MetaLeft') {
+      document.querySelector('.bill').style.display = 'block';
+    }
+
+    this.printedText.focus();
+
+    if (AllCodesAndKeys[event.code][0] === 'CapsLock') {
+      if (document.getElementById(`${event.code}`).classList.contains('active')) {
```

Comment:
just store `const button = document.getElementById(`${event.code}`)` and use later to avoid code duplication

Reply:
fixed